incubator/hnc/internal/validators/namespace.go

```diff
@@ -79,7 +79,26 @@ func (v *Namespace) checkForest(req *nsRequest) admission.Response {
 
 	ns := v.Forest.Get(req.ns.Name)
 
-	// Early exit to allow non-delete requests or if the namespace allows cascading deletion.
+	// Allow all create namespace requests except if the namespace name already exists in
+	// the external hierarchy (the default webhook already denies existing namespace names).
+	if req.op == v1beta1.Create {
+		for _, nm := range v.Forest.GetNamespaceNames() {
+			if _, ok := v.Forest.Get(nm).ExternalTreeLabels[req.ns.Name]; ok {
+				return deny(metav1.StatusReasonAlreadyExists, "The name is already used. Please us a different name.")
+			}
+		}
+	}
+
+	// Allow all update namespace requests except if the namespace has a parent and wants
+	// to add "hnc.x-k8s.io/managedBy" annotation other than "hnc.x-k8s.io".
+	if req.op == v1beta1.Update && req.ns.Annotations[api.AnnotationManagedBy] != "" {
+		if req.ns.Annotations[api.AnnotationManagedBy] != api.MetaGroup && ns.Parent() != nil {
+			msg := fmt.Sprintf("Namespace %s is a child of %s. Namespaces with parent cannot be converted to external namespaces. Please unset the parent first.", ns.Name(), ns.Parent().Name())
```

Comment:
Use %q instead of %s (I've found it reads much more easily). Also how about the following change?

> Namespace %q is a child of %q. Namespaces with parents defined by HNC cannot also be managed externally. To manage this namespace with %q, first make it a root in HNC.